editor/state.js

```diff
@@ -10,17 +10,35 @@ import { keyBy, last, omit, without } from 'lodash';
 import { combineUndoableReducers } from 'utils/undoable-reducer';
 
 /**
- * Reducer returning editor blocks state, an combined reducer of keys byUid,
- * order, where blocks are parsed from current HTML markup.
+ * Undoable reducer returning the editor post state, including blocks parsed
+ * from current HTML markup.
+ *
+ * Handles the following state keys:
+ *  - post: an object describing the current post, in the format used by the WP
+ *  REST API
+ *  - blocksByUid: post content blocks keyed by UID
+ *  - blockOrder: list of block UIDs in order
  *
  * @param  {Object} state  Current state
  * @param  {Object} action Dispatched action
  * @return {Object}        Updated state
  */
-export const blocks = combineUndoableReducers( {
-	byUid( state = {}, action ) {
+export const editor = combineUndoableReducers( {
+	post( state = {}, action ) {
+		switch ( action.type ) {
+			case 'EDIT_POST':
+				return action.post || state;
+
+			case 'POST_UPDATE_REQUEST_SUCCESS':
+				return action.post;
```

Comment:
Good point.  Probably the entire post object shouldn't be included in the undoable state, rather just the list of edits like we do in Calypso.  Seems closely related to the thread below as well (https://github.com/WordPress/gutenberg/pull/594#discussion_r114677900).